src/test/java/io/reactivex/maybe/MaybeTest.java

```diff
@@ -2997,4 +2997,15 @@ public boolean getAsBoolean() throws Exception {
             }
         }).test().assertResult(1);
     }
+
+    @Test
+    public void switchIfEmptySingle() {
+        Maybe.empty().switchIfEmpty(Single.just(1)).test().assertResult(1);
```

Comment:
The aim is to ensure `switchIfEmpty` returns `Single`. This fluent writing works if `switchIfEmpty` returned any other type.